Fix Java 11

JonasCir opened the pull request:
- update docs to mention Java 11
- make build for plain openjdk 11 pass
- upgrade CI pipeline to openjdk 11

fixes #2612

MartinWahnschaffeSymeda commented:
See https://github.com/hzi-braunschweig/SORMAS-Project/issues/2612#issuecomment-671345774

JonasCir commented:
@MartinWahnschaffeSymeda 

- Comment to docs added: Java 8 only (for now)
- CI builds for Java 8 + 11: Build runs in parallel, no impact on CI times expected
- Tracking issue added: #2617

JonasCir commented:
Wait for #2589 merge :D

JonasCir commented:
@MartinWahnschaffeSymeda 
Can you review and restart the CI? I addressed all comments and I think this PR is good to go. CI failure was caused by the flaky test you fixed :+1:

xavier-calland commented on DEVELOPMENT_ENVIRONMENT.md:
```diff
@@ -10,14 +10,14 @@
 		sudo apt-get update
-		sudo apt-get install zulu-8
-  - **Windows**: For testing and development environments we suggest to download and run the installer of the Java 8 **JDK** for 32 or 64 bit client systems (depending on your system).
+		sudo apt-get install zulu-11
```
It seems the package is now called `zulu11`

JonasCir replied:
I just tested this on a fresh ubuntu  20.04 LTS and the package name is `zulu-11`

xavier-calland commented on sormas-base/dev-setup/server-setup-dev-docker.sh:
```diff
@@ -28,9 +28,9 @@ echo "# Welcome to the SORMAS server setup routine. This script will guide you t
-#AS_JAVA_NATIVE=/usr/lib/jvm/zulu-8-amd64/
+#AS_JAVA_NATIVE='C:\zulu-11'
+#AS_JAVA_NATIVE='/opt/zulu-11'
+#AS_JAVA_NATIVE=/usr/lib/jvm/zulu-11-amd64/
```
and the path is now `/usr/lib/jvm/zulu11/`

JonasCir replied:
I just tested this on a fresh ubuntu  20.04 LTS and the path is `/usr/lib/jvm/zulu-11-amd64/`

JonasCir commented:
@xavier-calland Thanks for the review! I ran a test on a fresh ubuntu image and the package name and the path worked. What system/what package manager do you use?

xavier-calland commented:
I'm using Ubuntu 20.04 too.

Both `zulu11` and `zulu-11` packages are available. `zulu11` is the new package name.


`apt show zulu-11`
````
Package: zulu-11
Version: 11.99+0-1
Priority: extra
Section: **oldlibs**
Maintainer: Azul Systems, Inc. <azul_openjdk@azulsystems.com>
Installed-Size: 26,6 kB
Depends: zulu11
Homepage: http://www.azul.com/zulu
Download-Size: 2 968 B
APT-Sources: https://repos.azul.com/zulu/deb stable/main amd64 Packages
Description: Transitional package for the Azul Systems Zulu JDK
 **This is a transitional package to move Azul Systems Zulu packages to the new repository location.**
````

`apt show zulu11`

````
Package: zulu11
Version: 11.0.8-1
Priority: extra
Section: java
Maintainer: Azul Systems, Inc. <azul_openjdk@azulsystems.com>
Installed-Size: 25,6 kB
Depends: zulu11-jdk (= 11.0.8-1), zulu11-ca (= 11.0.8-1)
Homepage: http://www.azul.com/zulu
Download-Size: 2 884 B
APT-Manual-Installed: yes
APT-Sources: https://repos.azul.com/zulu/deb stable/main amd64 Packages
Description: Azul Systems Zulu 11 JDK
 Azul Zulu is an enterprise-quality, commercialized distribution of OpenJDK that
 can be deployed across various operating systems, Docker containers, hypervisors
 and Cloud platforms.
 Zulu is a compliant implementation of the Java SE specification, verified with
 the OpenJDK Community Technology Compatibility Kit.
 Zulu distributions are available for a variety of Linux operating systems with
 appropriate TGZ, APK, RPM, and Debian package types, including Linux repository
 access via yum, apk, and apt-get.
 To see all Zulu details, review user documentation, and check for downloads,
 visit https://www.azul.com/zulu.
````

JonasCir commented:
@xavier-calland I found the difference: SERVER_SETUP.md and DEVELOPMENT_ENVIRONMENT.md add the PPA with `deb http://repos.azulsystems.com/ubuntu stable main`. However, your log out put says:
`https://repos.azul.com/zulu/deb stable/main amd64`
If I'm not mistaken, you use a different PPA.

xavier-calland commented:
@JonasCir Apparently Azul has (recently) changed the organization of APT repository.
The [documentation](http://docs.azul.com/zulu/zuludocs/ZuluUserGuide/PrepareZuluPlatform/AttachAPTRepositoryUbuntuOrDebianSys.htm) talks about https://cdn.azul.com/zulu/bin/zulu-repo_1.0.0-2_all.deb which adds the `/etc/apt/sources.list.d/zulu-openjdk.list` file with `deb [arch = amd64] https://repos.azul.com/zulu/deb/ stable main`

JonasCir commented:
@xavier-calland moved to the new zulu11 package repo, CI is green now, so I guess this is good to go

xavier-calland commented:
@JonasCir , looks good to me

JonasCir commented:
@xavier-calland thanks for your help! Do have the permissions to merge this? Otherwise @MartinWahnschaffeSymeda maybe? :smiley:

MartinWahnschaffeSymeda commented:
We are currently doing a release. I'll merge this afterwards.
